front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -258,155 +263,376 @@ const createTrainSchedulePayload = async ({
+  }
+  const newPacedTrain: PacedTrainResponseWithPacedTrainId = {
+    ...newTimetableItems[0],
+    id: formatEditoastTrainIdToPacedTrainId(newTimetableItems[0].id),
   };
+  state.timetableItemIdByNgeId.set(trainrun.id, newPacedTrain.id);
+  addUpsertedTimetableItems([newPacedTrain]);
 };
 
-const handleUpdateTrainSchedule = async ({
+const handleUpdateTimetableItem = async ({
+  timetableId,
   netzgrafikDto,
   trainrun,
   dispatch,
   infraId,
-  addUpsertedTrainSchedules,
+  addUpsertedTimetableItems,
   state,
 }: {
+  timetableId: number;
   netzgrafikDto: NetzgrafikDto;
   trainrun: TrainrunDto;
   dispatch: AppDispatch;
   infraId: number;
-  addUpsertedTrainSchedules: (trainSchedules: TrainScheduleResultWithTrainId[]) => void;
+  addUpsertedTimetableItems: (timetableItems: TimetableItemWithTimetableId[]) => void;
   state: MacroEditorState;
-}) => {
-  const { nodes, labels } = netzgrafikDto;
-  const trainrunSectionsByTrainrunId = getTrainrunSectionsByTrainrunId(netzgrafikDto, trainrun.id);
-  // TODO Paced train : Adapt to handle paced trains in issue https://github.com/OpenRailAssociation/osrd/issues/10612
-  const editoastTrainScheduleIdToUpdate = formatTrainScheduleIdToEditoastTrainId(
-    state.trainScheduleIdByNgeId.get(trainrun.id) as TrainScheduleId
+}): Promise<void> => {
+  const timetableItem = await fetchTimetableItem(
+    state.timetableItemIdByNgeId.get(trainrun.id)!,
+    dispatch
   );
-  const trainSchedule = await dispatch(
-    osrdEditoastApi.endpoints.getTrainScheduleById.initiate({
-      id: editoastTrainScheduleIdToUpdate,
-    })
-  ).unwrap();
-  const startDate = new Date(trainSchedule.start_time);
-  const trainSchedulePayload = await createTrainSchedulePayload({
-    trainrunSections: trainrunSectionsByTrainrunId,
-    nodes,
+  const { path, trainrunLabels, startDate, schedule } = await generateTrainrunData(
+    getTrainrunSectionsByTrainrunId(netzgrafikDto, trainrun.id),
+    netzgrafikDto.nodes,
     trainrun,
     infraId,
     dispatch,
-    labels,
-    oldStartDate: startDate,
-    trainSchedule,
-  });
-  const newTrainSchedule = await dispatch(
-    osrdEditoastApi.endpoints.putTrainScheduleById.initiate({
-      id: editoastTrainScheduleIdToUpdate,
-      trainScheduleForm: {
-        ...trainSchedule,
-        ...trainSchedulePayload,
-        // Reset margins because they contain references to path items
-        margins: undefined,
-      },
+    netzgrafikDto.labels,
+    timetableItem.start_time
+  );
+  const newTimetableItem = generateTimetableItem(
+    trainrun,
+    trainrunLabels,
+    path,
+    startDate,
+    schedule,
+    timetableItem,
+    state
+  );
+  const updatedTimetableItem = await updateTimetableItem(
+    timetableId,
+    {
+      ...timetableItem,
+      ...newTimetableItem,
+    },
+    getFrequencyFromFrequencyId(state, trainrun.frequencyId),
+    dispatch
+  );
+  addUpsertedTimetableItems([updatedTimetableItem]);
```

Comment:
I think you don't need to get updatedTimetableItem from updateTimetableItem, since you already have it (it's what you send to the backend !). (do you really need this newTimetableItemBis ? newTimetableItem is not enough ?)
```suggestion
  const newTimetableItemBis =  {
      ...timetableItem,
      ...newTimetableItem,
    }
  await updateTimetableItem(
    timetableId,
    newTimetableItemBis,
    getFrequencyFromFrequencyId(state, trainrun.frequencyId),
    dispatch
  );
  addUpsertedTimetableItems([newTimetableItemBis]);
```

Reply:
Solved by "refactor generateTimetableItem" i think